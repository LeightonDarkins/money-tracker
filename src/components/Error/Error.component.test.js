/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ErrorComponent from './Error.component'
import th from '../../common/TestHelpers'

describe('Error', () => {
  describe('Snapshots', () => {
    it('should render an empty ErrorHandler with no errors', () => {
      const errorComponent = renderer.create(<ErrorComponent
        title={'title'}
        message={'message'}
        onClick={th.empty}
        />).toJSON()

      expect(errorComponent).toMatchSnapshot()
    })
  })

  describe('Behaviour', () => {
    let mockOnClick

    beforeEach(() => {
      mockOnClick = jest.fn()
    })

    afterEach(() => {
      mockOnClick.mockReset()
    })

    it('calls onErrorClick when an Error is clicked', () => {
      let wrapper = shallow(<ErrorComponent
        title={'title'}
        message={'message'}
        onClick={mockOnClick}
        />)

      wrapper.simulate('click')

      expect(mockOnClick).toBeCalled()
    })
  })
})
