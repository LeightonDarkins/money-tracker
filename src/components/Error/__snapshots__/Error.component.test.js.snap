exports[`Error Snapshots should render an empty ErrorHandler with no errors 1`] = `
<div
  className="ErrorComponent"
  onClick={[Function]}>
  <h3
    className="heading">
    title
  </h3>
  <p
    className="body">
    message
  </p>
  <i
    aria-hidden="true"
    className="fa fa-times close" />
</div>
`;
